File: 78_subsets.py
from itertools import combinations
from typing import List


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        N = len(nums)
        ans = [[]]

        for i in range(1, N + 1):
            ans.extend(combinations(nums, i))

        return ans

File: 78_subsets.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        
        for (int mask = 0; mask < (1<<nums.size()); mask++) {
            vector<int> sub;
            
            for (int j = 0; j < nums.size(); j++) {
                if (mask & (1<<j)) {
                    sub.push_back(nums[j]);
                }
            }
            
            res.push_back(sub);
        }
        
        return res;
    }
};